examples/example_analysis_workflows.py

```diff
@@ -0,0 +1,197 @@
+"""Example high-level analysis workflows.
+
+This code doesn't run yet, it is just the result of some brainstorming how we
+could structure analysis functionality in Gammapy.
+
+The basic idea is to have "analysis" classes for the following cases:
+- 0-dim region-based analysis
+- 1-dim spectrum analysis
+- 2-dim image analysis
+- 3-dim cube analysis
+
+Typically the analysis will be split into the following classes:
+
+- SpectrumMaker, ImageMaker, CubeMaker
+  - Computes counts and exposure and sometimes background arrays
+- SpectrumObservation, ImageObservation, CubeObservation
+  - Container with counts, exposure and background arrays
+  - Has extra info like PSF, depending on the analysis
+  - Filled by the maker classes
+- SpectrumFitter, ImageFitter, CubeFitter
+  - Takes lists of observation objects as input
+  - Those lists could be Python lists or our own SpectrumObservationList etc,
+    with convenience methods attached (like print summary, plot summary, ...)
+
+These classes could be located in (to be discussed)
+
+- SpectrumMaker, SpectrumObservation and SpectrumFitter in gammapy.spectrum and similarly for image and cube
+- Or, have the container classes SpectrumObservation and SpectrumObservationList in spectrum
+  and the "analysis" or "maker" classes in `gammapy.scripts` (which should be thought of as high-level,
+  end-user pre-baked common analysis functionality, whether it's exposed via command line interface or not.
+
+To be discussed:
+
+- When are results attached to analysis or container classes?
+  And when split out into separate "results" classes?
+- At the moment we use "image" and "map" to mean the same thing.
+  Should we commit to one term?
+"""
+
+
+def get_observations():
+    """Get `ObservationList`, i.e. the IACT dataset to analyse.
+
+    This will be the common first step for all the IACT analyses.
+
+    The idea is that `ObservationList` is the data definition / access object.
+    Data is loaded from disk into memory on access.
+
+    TODO:
+    - When is data discarded from memory?
+    - Should the observation grouping specification happen here,
+      so that the obs grouping code isn't duplicated for the various analyses?
+    """
+    from gammapy.data import DataStore, ObservationList
+    data_store = DataStore.from_dir(dir=...)
+    obs_id = data_store.obs_table.cone_select(...)
+
+    observations = ObservationList(data_store=..., obs_id=...)
```

Comment:
It seem a little weird to overwrite the ``list`` ``__init__`` method. What about

```
observations = ObservationList()
observations.add_obs_from_store(data_store=..., obs_id=....)
```

Then, this would also work

```
observations = ObservationList([data_store.obs(_) for _ in obs_ids])

```
```